stress: failed test in cockroach/kv/kv.test: TestPropagateTxnOnPushError

The following test appears to have failed:

```
=== RUN   TestPropagateTxnOnPushError
I160306 05:05:03.896031 server/node.go:353  store store=0:0 ([]=) not bootstrapped
W160306 05:05:03.897368 gossip/gossip.go:919  not connected to cluster; use --join to specify a connected node
I160306 05:05:03.897422 storage/replica_command.go:1255  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1457240704.897s
I160306 05:05:03.897697 server/node.go:299  **** cluster cde14a3d-9932-4cd3-8c7f-f9c6054b5e40 has been created
I160306 05:05:03.897712 server/node.go:300  **** add additional nodes by specifying --join=127.0.0.1:33026
I160306 05:05:03.898010 server/node.go:366  initialized store store=1:1 ([]=): {Capacity:8312655872 Available:6110269440 RangeCount:0}
I160306 05:05:03.898027 server/node.go:275  node ID 1 initialized
I160306 05:05:03.898058 storage/stores.go:289  read 0 node addresses from persistent storage
I160306 05:05:03.898102 server/node.go:486  connecting to gossip network to verify cluster ID...
I160306 05:05:03.898489 server/node.go:507  node connected via gossip and verified as part of cluster "cde14a3d-9932-4cd3-8c7f-f9c6054b5e40"
I160306 05:05:03.898512 server/node.go:331  [node=1] Started node with [[]=] engine(s) and attributes []
I160306 05:05:03.898528 server/server.go:315  starting https/postgres server at 127.0.0.1:33026
I160306 05:05:03.899059 server/status/runtime.go:137  runtime stats: 37 goroutines, 3.94MiB active, 0.00cgo/sec, 0.00/0.00 %(u/s)time, 0.00 %gc (271x)
I160306 05:05:03.899228 storage/split_queue.go:97  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0]
I160306 05:05:03.921765 storage/replica_command.go:1332  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160306 05:05:03.927500 storage/replica_command.go:1255  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1457240704.948s
I160306 05:05:03.927612 storage/replica_command.go:1332  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160306 05:05:03.940232 storage/replica_command.go:1255  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1457240704.948s
I160306 05:05:03.940320 storage/replica_command.go:1332  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
W160306 05:05:03.943798 storage/replica.go:1781  [node=1,store=1,range=1] failed to push during intent resolution: conflicting intents on [/Meta2/Table/13 /Meta2/Max]: resolved? false
I160306 05:05:03.989704 storage/replica_command.go:1332  initiating a split of range=1 [/Min-/Table/11) at key "b"
I160306 05:05:03.991994 storage/replica_command.go:1255  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1457240704.992s
W160306 05:05:03.992442 storage/replica.go:1781  [node=1,store=1,range=1] failed to push during intent resolution: conflicting intents on [/Meta1/Max]: resolved? false
I160306 05:05:03.995458 storage/replica_command.go:1255  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1457240704.995s
I160306 05:05:03.997929 stopper.go:273  draining; tasks left:
1      storage/replica.go:1832
1      server/node.go:689
1      kv/txn_coord_sender.go:823
I160306 05:05:03.997974 stopper.go:273  draining; tasks left:
1      storage/replica.go:1832
1      server/node.go:689
--- FAIL: TestPropagateTxnOnPushError (5.11s)
	dist_sender_server_test.go:612: expected TransactionRetryError, but got: <nil>
	dist_sender_server_test.go:621: unexpected epoch; the txn must be attempted 2 times, but got 1 attempts
	leaktest.go:80: Leaked goroutine: goroutine 9040 [chan receive]:
		github.com/cockroachdb/cockroach/kv_test.TestPropagateTxnOnPushError.func1.1(0xc82023e000, 0xc8206b26f4)
			/go/src/github.com/cockroachdb/cockroach/kv/dist_sender_server_test.go:563 +0x13a
		github.com/cockroachdb/cockroach/client.(*DB).Txn.func1(0xc82023e000, 0xc8206b26f4, 0x4)
			/go/src/github.com/cockroachdb/cockroach/client/db.go:468 +0x25
		github.com/cockroachdb/cockroach/client.(*Txn).Exec(0xc82023e000, 0x1be0101, 0xc82052ae58, 0x1)
			/go/src/github.com/cockroachdb/cockroach/client/txn.go:462 +0x394
		github.com/cockroachdb/cockroach/client.(*DB).Txn(0xc82085d8b0, 0xc82052af68, 0x0)
			/go/src/github.com/cockroachdb/cockroach/client/db.go:469 +0x1a1
		github.com/cockroachdb/cockroach/kv_test.TestPropagateTxnOnPushError.func1(0xc82085d8b0, 0xc820876120, 0xc820876180, 0xc82077e000, 0xc8208761e0)
			/go/src/github.com/cockroachdb/cockroach/kv/dist_sender_server_test.go:556 +0x62
		created by github.com/cockroachdb/cockroach/kv_test.TestPropagateTxnOnPushError
			/go/src/github.com/cockroachdb/cockroach/kv/dist_sender_server_test.go:569 +0x247
FAIL


ERROR: exit status 1
```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender_server_test.go
+++ b/kv/dist_sender_server_test.go
@@ -547,16 +547,23 @@ func TestPropagateTxnOnPushError(t *testing.T) {
 	waitForWriteIntent := make(chan struct{})
 	waitForTxnRestart := make(chan struct{})
 	waitForTxnCommit := make(chan struct{})
+	lowPriority := int32(1)
+	highPriority := int32(10)
+	key := "a"
 	// Create a goroutine that creates a write intent and waits until
 	// another txn created in this test is restarted.
 	go func() {
 		if pErr := db.Txn(func(txn *client.Txn) *roachpb.Error {
-			if pErr := txn.Put("b", "val"); pErr != nil {
+			// Set high priority so that the intent will not be pushed.
+			txn.InternalSetPriority(highPriority)
+			log.Infof("Creating a write intent with high priority")
+			if pErr := txn.Put(key, "val"); pErr != nil {
 				return pErr
 			}
 			close(waitForWriteIntent)
 			// Wait until another txn in this test is
 			// restarted by a push txn error.
+			log.Infof("Waiting for the txn restart")
 			<-waitForTxnRestart
 			return txn.CommitInBatch(txn.NewBatch())
 		}); pErr != nil {
@@ -566,39 +573,34 @@ func TestPropagateTxnOnPushError(t *testing.T) {
 	}()
 
 	// Wait until a write intent is created by the above goroutine.
+	log.Infof("Waiting for the write intent creation")
 	<-waitForWriteIntent
 
-	// The transaction below is restarted multiple times.
-	// - The first retry is caused by the write intent created on key "b" by the above goroutine.
-	// - The subsequent retries are caused by the write conflict on key "a". Since the txn
-	//   ID is not propagated, a txn of a new epoch always has a new txn ID different
-	//   from the previous txn's. So, the write intent made by the txn of the previous epoch
-	//   is treated as a write made by some different txn.
+	// The transaction below is restarted exactly once. The restart is
+	// caused by the write intent created on key "a" by the above goroutine.
+	// When the txn is retried, the error propagates the txn ID to the next
+	// iteration.
 	epoch := 0
 	var txnID *uuid.UUID
 	if pErr := db.Txn(func(txn *client.Txn) *roachpb.Error {
-		// Set low priority so that the intent will not be pushed.
-		txn.InternalSetPriority(1)
+		// Set low priority so that a write from this txn will not push others.
+		txn.InternalSetPriority(lowPriority)
 
 		epoch++
 
 		if epoch == 2 {
 			close(waitForTxnRestart)
 			// Wait until the txn created by the goroutine is committed.
+			log.Infof("Waiting for the txn commit")
 			<-waitForTxnCommit
 			if !roachpb.TxnIDEqual(txn.Proto.ID, txnID) {
 				t.Errorf("txn ID is not propagated; got %s", txn.Proto.ID)
 			}
 		}
 
-		b := txn.NewBatch()
-		b.Put("a", "val")
-		b.Put("b", "val")
-		// The commit returns an error, but it will not be
-		// passed to the next iteration. txnSender.Send() does
-		// not update the txn data since
-		// TransactionPushError.Transaction() returns nil.
-		pErr := txn.CommitInBatch(b)
+		// The commit returns an error, and it will pass
+		// the txn data to the next iteration.
+		pErr := txn.Put(key, "val")
 		if epoch == 1 {
 			if tErr, ok := pErr.GetDetail().(*roachpb.TransactionPushError); ok {
 				if pErr.GetTxn().ID == nil {
@@ -616,5 +618,11 @@ func TestPropagateTxnOnPushError(t *testing.T) {
 
 	if e := 2; epoch != e {
 		t.Errorf("unexpected epoch; the txn must be attempted %d times, but got %d attempts", e, epoch)
+		if epoch == 1 {
+			// Wait for the completion of the goroutine to see if it successfully commits the txn.
+			close(waitForTxnRestart)
+			log.Infof("Waiting for the txn commit")
+			<-waitForTxnCommit
+		}
 	}
 }

